api/tests/unit/task_processor/conftest.py

```diff
@@ -1,6 +1,35 @@
+import logging
+import typing
+
 import pytest
 
 
 @pytest.fixture
 def run_by_processor(monkeypatch):
     monkeypatch.setenv("RUN_BY_PROCESSOR", "True")
+
+
+class GetTaskProcessorCaplog(typing.Protocol):
+    def __call__(
+        self, log_level: str | int = logging.INFO
+    ) -> pytest.LogCaptureFixture: ...
+
+
+@pytest.fixture
+def get_task_processor_caplog(
+    caplog: pytest.LogCaptureFixture,
+) -> GetTaskProcessorCaplog:
+    # caplog doesn't allow you to capture logging outputs from loggers that don't
+    # propagate to root. Quick hack here to get the task_processor logger to
+    # propagate.
+    # TODO: look into using loguru.
+
```

Comment:
@matthewelwell Why doesn't it propagate in the first place?

Reply:
TBH, I'm not sure but I don't really want to open that can of worms here. [Context](https://github.com/Flagsmith/flagsmith/blob/main/api/app/settings/common.py#L564-L588).